### PR TITLE
Import Docker build infrastructure from jellyfin-android

### PR DESCRIPTION
**Pull request by joshuaboniface**
Imports the Docker-based APK build facilities from [jellyfin-android](https://github.com/jellyfin/jellyfin-android). Combined with the `gradlew` configuration from #37 generates APKs in a similar format and set of release types (production, unminified, and debug).

### CODE REVIEW

File: docker-build.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+# Handle the release type
+case ${RELEASE} in
+    'production')
+        RELEASE_SUFFIX=""
+        NODE_ENV="production"
+        RFLAG="assembleRelease"
+    ;;
+    'unminified')
+        RELEASE_SUFFIX="unminified_"
```
Comment (dkanada) [inline docker-build.sh]: We don't use node so this section won't do anything, need to find a new way to specify debug as well.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# Docker run environment
+ENV SOURCE_DIR=/repo
+ENV ARTIFACT_DIR=/dist
+ENV ANDROID_DIR=/usr/lib/android-sdk
+
+# Prepare Debian build environment
+RUN echo "deb http://deb.debian.org/debian stretch-backports main" >> /etc/apt/sources.list.d/backports.list \
+ && apt-get update \
+ && apt-get install -y mmv wget unzip git \
+ && apt-get install -t stretch-backports -y npm android-sdk openjdk-8-jdk
```
Comment (thornbill) [inline Dockerfile]: `npm` is not needed here.

File: docker-build.sh (hunk 1 above)
Comment (joshuaboniface) [reply docker-build.sh]: The `assembleDebug` does that, or do you mean specifying unminified?

File: Dockerfile (hunk 2 above)
Comment (joshuaboniface) [reply Dockerfile]: Good catch, no sense wasting the space.

File: docker-build.sh (hunk 1 above)
Comment (dkanada) [reply docker-build.sh]: specifying unminified

Comment (joshuaboniface) [reply docker-build.sh]: As discussed on Riot, might be OK to just not have this as this app seems to not have as many issues with minification. Worth research for the future though.

File: .gitignore [hunk 3]
```diff
@@ -4,4 +4,4 @@ local.properties
 
 /.idea
 /.gradle
-build
+gradle
```
Comment (thornbill) [inline .gitignore]: I don’t think this should be added back. Merge conflict maybe?

Comment (joshuaboniface) [reply .gitignore]: Ah yes I mis-corrected that merge conflict, I'll remove this too.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+RUN echo "deb http://deb.debian.org/debian stretch-backports main" >> /etc/apt/sources.list.d/backports.list \
+ && apt-get update \
+ && apt-get install -y mmv wget unzip git \
+ && apt-get install -t stretch-backports -y android-sdk openjdk-8-jdk
+
+# Install Android repository tools
+RUN rm -rf ${ANDROID_DIR}/tools \
+ && wget http://dl-ssl.google.com/android/repository/tools_r25.2.3-linux.zip -O tools.zip \
+ && unzip tools.zip -d ${ANDROID_DIR}/ \
+ && rm -f tools.zip
```
Comment (dkanada) [inline Dockerfile]: I still think we should be able to remove this somehow but that can be for another day.

Comment (joshuaboniface) [reply Dockerfile]: As long as we need to use sdkmanager, we need it :(

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+# Install SDK tools
+# `yes` is required to accept licenses:
+# https://stackoverflow.com/questions/38096225/automatically-accept-all-sdk-licences/42125740#42125740
+# There will be a bit of delay (approx 30 seconds)
```
Comment (LeoVerto) [inline Dockerfile]: I know I'm repeating myself but this should be something along the lines of "up to a couple minutes".

Comment (joshuaboniface) [reply Dockerfile]: Agreed, I updated the wording.

**Pull request merged**